boards: arm: Add support for ARM FK7B0M1-VBT6 V1.0 board

CharlesDias opened the pull request:
The following features have been tested and confirmed to work on the hardware:

- LED
- Button
- UART
- Flashing the binary via `west flash` command

kartben commented on boards/arm/fk7b0m1_vbt6/doc/index.rst:
```diff
@@ -0,0 +1,185 @@
+     :align: center
+     :alt: FK7B0M1-VBT6
+
+     FK7B0M1-VBT6 (Credit: Charles Dias)
```
Is the credit really yourself? I seem to find a very similar picture online. FWIW it's ok to reuse a photo from the vendor as it falls under fair use. You may also consider using webp format for further compression/quality and support of transparent background.

CharlesDias replied:
Sorry! My bad. I'll fix it.

kartben commented on boards/arm/fk7b0m1_vbt6/doc/index.rst:
```diff
@@ -0,0 +1,185 @@
+     :align: center
+     :alt: FK7B0M1-VBT6
+
+     FK7B0M1-VBT6 (Credit: Charles Dias)
```
Ditto re: image credit

CharlesDias replied:
Done.

kartben commented on boards/arm/fk7b0m1_vbt6/fk7b0m1_vbt6.dts:
```diff
@@ -0,0 +1,97 @@
+
+/ {
+	model = "FANKE FK7B0M1-VBT6 board V1.0";
+	compatible = "fanke,fk7b0m1-vbt6";
```
This vendor prefix needs to be added to https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/vendor-prefixes.txt

CharlesDias replied:
Done.

kartben commented on boards/arm/fk7b0m1_vbt6/doc/index.rst:
```diff
@@ -0,0 +1,185 @@
+The FK7B0M1-VBT6 board does not include an on-board debugger. As a result, it requires
+an external debugger, such as ST-Link, for programming and debugging purposes.
+
+The board provides a header pins for the Serial Wire Debug (SWD) interface.
```
```suggestion
The board provides header pins for the Serial Wire Debug (SWD) interface.
```

CharlesDias replied:
Done.

CharlesDias replied:
Thanks, @kartben!

CharlesDias commented on boards/arm/fk7b0m1_vbt6/Kconfig.defconfig:
```diff
@@ -0,0 +1,11 @@
+# STM32H7B0VBT board board configuration
```
Done.

CharlesDias commented on boards/arm/fk7b0m1_vbt6/board.cmake:
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(jlink "--device=STM32H7B0VB" "--speed=4000")
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
+board_runner_args(openocd --target-handle=_CHIPNAME.cpu0)
+
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
```
Done.

CharlesDias commented:
Hi @kartben and @galak. Any other changes?

kartben commented on boards/arm/fk7b0m1_vbt6/board.cmake:
```diff
@@ -0,0 +1,9 @@
+
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
```
missing a newline at the end of the file (not sure why CI doesn't catch this)

CharlesDias replied:
Done. Let me know if any other modification is needed.

kartben reviewed (changes requested):
Documentation looks good, thanks! A few extra comments on my side + you should also please squash the documentation commit with the one introducing the board, as new boards are typically added as just one commit

CharlesDias committed:
boards: arm: Add support for FK7B0M1-VBT6 board

Adds the device trees, Kconfig, and documentation files.

The following features have been confirmed working on hardware:
* LED
* Button
* UART

Signed-off-by: Charles Dias <charlesdias.cd@outlook.com>

CharlesDias committed:
dts: bindings: Add vendor prefix fanke

Add vendor prefix fanke FANKE Technology Co., Ltd.

Signed-off-by: Charles Dias <charlesdias.cd@outlook.com>

CharlesDias commented on boards/arm/fk7b0m1_vbt6/doc/index.rst:
```diff
@@ -0,0 +1,185 @@
+
+     FK7B0M1-VBT6 (Credit: FANKE Technology Co., Ltd)
+
+
```
Done.

erwango commented on boards/arm/fk7b0m1_vbt6/doc/index.rst:
```diff
@@ -0,0 +1,179 @@
+.. _fk7b0m1_vbt6:
+
+FANKE FK7B0M1-VBT6 board V1.0
```
Open question:

Zephyr board description model supports versioning. Hence, follow up revisions of this boards could theoretically be available under this same directory and also documentation.

Should we keep the "V1.0" mention here ?

CharlesDias replied:
Hi, @erwango.

In this case, do I need to remove the "V1.0 board" and create the **revision.cmake** file with the content `board_check_revision(FORMAT MAJOR.MINOR.PATCH DEFAULT_REVISION 0.0.0)` ?

Since the 1.0 is the only version that currently exists, should I also create the **fk7b0m1_vbt6_1_0_0.conf** and **fk7b0m1_vbt6_1_0_0.overlay** as an empty file?  Or does those files only need when another version was released?

Thanks for your guidance.

erwango replied:
If this is the only one that exists today, I don't think you have to bother.
You can add this later on when (/if) a v1.X pops up.

CharlesDias replied:
I removed the "V1.0".

FRASTM reviewed (commented):
else LGTM

kartben commented on boards/arm/fk7b0m1_vbt6/doc/index.rst:
```diff
@@ -0,0 +1,179 @@
+.. _fk7b0m1_vbt6:
+
+FANKE FK7B0M1-VBT6 board
+##############################
```
```suggestion
FANKE FK7B0M1-VBT6 board
########################
```

kartben replied:
remove extra #####s, but also probably can do without "... board"?

```suggestion
FANKE FK7B0M1-VBT6
##################
```

kartben commented on boards/arm/fk7b0m1_vbt6/support/openocd.cfg:
```diff
@@ -0,0 +1,25 @@
+# connect_assert_srst needed if low power mode application running (WFI...)
+reset_config srst_only srst_nogate connect_assert_srst
+set CONNECT_UNDER_RESET 1
+set CORE_RESET 0
```
adding an empty line at the end of the file would be nice.

kartben reviewed (commented):
thanks for addressing my earlier comments, just spotted a couple extra (and minor) issues. Thanks!
